compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -880,6 +880,16 @@ object messages {
            |"""
   }
 
+  case class UncheckedTypePattern(msg: String)(implicit ctx: Context)
+    extends Message(UncheckedTypePatternID) {
+    val kind = "Unchecked Type Pattern"
```

Comment:
This is a very specific kind, kinds are supposed to apply to more than one message, I guess here it could be "Pattern Match Exhaustivity" (kinds should also be an enum, but that's a separate issue).